gap/NautyGraph.gi

```diff
@@ -361,6 +361,40 @@ InstallMethod( CanonicalForm,
     
 end );
 
+InstallMethod( CanonicalForm,
+               [ IsNautyEdgeColoredGraphRep ],
+               
+  function( graph )
+    local colors, edges, perm, new_graph, permEdges, i,
+        permInv, edgeClass, newClass;
+    
+    perm := CanonicalLabeling( graph );
+    permInv := CanonicalLabelingInverse( graph );
+    
+    edges := graph!.edge_list;
+
+    permEdges := [];
+    # Since this computation should be efficient we do it manually
+    for edgeClass in edges do
+        newClass := [];
+
+        for i in [ 1 .. Length( edgeClass ) ] do # edges is a dense list
+            # Every line of code in this loop costs a lot of time
+            Add( newClass, [ edgeClass[ i ][ 1 ]^permInv, edgeClass[ i ][ 2 ]^permInv ] );
+            if newClass[ i ][ 1 ] > newClass[ i ][ 2 ] then
+                newClass[ i ] := [ newClass[ i ][ 2 ], newClass[ i ][ 1 ] ];
+            fi;
+        od;
+
+        Add(permEdges, Set(newClass) );
+    od;
+    edges := permEdges;
```

Comment:
Or you could do this instead:
```suggestion
    edges := List( graph!.edge_list, edgeClass -> OnSetsSets( edgeClass, permInv ) );
```

Reply:
Yes, but that would be slower. Since this loop is called many times, code that computes many isomorphisms is significantly speed up by the manual computation.

Reply:
Have you timed it? In any case, if you are concerned about performance, you should still consider replacing the body of the innermost loop by using `OnSets`, which is a kernel function and hard to beat with plain GAP code.